src/components/hmi_message_handler/src/hmi_message_handler_impl.cc

```diff
@@ -94,6 +95,7 @@ void HMIMessageHandlerImpl::AddHMIMessageAdapter(HMIMessageAdapter* adapter) {
     LOG4CXX_WARN(logger_, "HMIMessageAdapter is not valid!");
     return;
   }
+  sync_primitives::AutoLock lock(message_adapters_locker_);
```

Comment:
@dev-gh this lock is used in `HMIMessageHandlerImpl::SubscribeToHMINotification` which is RC-related. Also this lock was added in terms of [this PR](https://github.com/smartdevicelink/sdl_core/pull/1245) to fix segfault in related UT.